mypy/plugins/attrs.py

```diff
@@ -90,10 +100,16 @@ def argument(self, ctx: 'mypy.plugin.ClassDefContext') -> Argument:
                 if types:
                     args = UnionType.make_simplified_union(types)
                     init_type = ctx.api.anal_type(args)
+            if self.converter.is_attr_converters_optional and init_type:
+                # If the converter was attr.converter.optional(type) then add None to
+                # the allowed init_type.
+                args = UnionType.make_simplified_union([init_type, NoneTyp()])
+                init_type = ctx.api.anal_type(args)
```

Comment:
Why do you need the `anal_type` call here?

Reply:
Oh I guess by this point the type has been analyzed already.  I'll remove it.